Doc fixes (#7306)
solo-io/gloo: docs/content/guides/dev/building-and-deploying-gloo.md

weight: 6
---

You can build and deploy Gloo Edge Open Source from the source code.

## Before you begin

> 1. Follow the [setup guide](https://docs.solo.io/gloo-edge/latest/guides/dev/setting-up-dev-environment/) to clone the Gloo repository and install the project dependencies.

Reviewer: ```suggestion
1. Follow the [setup guide]({{% versioned_link_path fromRoot="/guides/dev/setting-up-dev-environment/" %}}) to clone the Gloo repository and install the project dependencies.
```

2. In your terminal, navigate to the Gloo project.
   ```sh
   cd ${GOPATH}/src/github.com/solo-io/gloo
   ```

> 3. Continue with this guide to learn how you can build all Gloo from the top-level [Makefile](https://github.com/solo-io/gloo/blob/master/Makefile).

Reviewer: ```suggestion
3. Continue with this guide to learn how you can build Gloo Edge from the top-level [Makefile](https://github.com/solo-io/gloo/blob/master/Makefile).
```

## Set up build dependencies and code generation {#setup}

> From the Gloo project directory, run:

Reviewer: add code ticks around the following command

    make -B install-go-tools generated-code

The `install-go-tools` target installs the dependencies to build Gloo in `_output/.bin`. 

The `generated-code` target does multiple things:

> 1. It generates the Go source code from the `.proto` files in the `api` directories. For example, `.proto` files in `projects/gloo/api` will be generated in `projects/gloo/pkg/api`

Reviewer: ```suggestion
1. It generates the Go source code from the `.proto` files in the `api` directories. For example, `.proto` files in `projects/gloo/api` will be generated in `projects/gloo/pkg/api`.
```

3. It generates [`solo-kit`](https://github.com/solo-io/solo-kit) resources, event loops, emitters, snapshots, and resource clients. These tools are denoted by `.sk.go`, and are built using `solo-kit.json` configuration files.

## Build Gloo and Docker Images {#build}

The Gloo project has several products. The code for `gloo` itself can be found in `projects/gloo`. Each product has its own set of build targets. 

**Gloo binary**

To compile the `gloo` binary to the `_output/projects/gloo/` directory, run:

    make -B gloo

**Docker image**

To build the Docker image, run the following command. Review the following table to understand the command options.

    make -B VERSION=0.0.1 IMAGE_REPO=localhost:5000 gloo-docker

| Option | Description |
| ------ | ----------- |
| `VERSION` | An optional version number for the Docker image tag, such as `0.0.1`. The format *must* be valid [semantic versioning](https://semver.org/). |
| `IMAGE_REPO` | The image repository for the image, such as the local host. This value is *required* on `arm64` or `m1` machines. |
| `gloo-docker` | The name for the Docker image. |

Example output:

    Successfully tagged quay.io/solo-io/gloo:0.0.1

> ## Deploy Gloo {#deploy}

Reviewer: I think we need to specify gllo edge to not confuse w gloo platform where possible
```suggestion
## Deploy Gloo Edge {#deploy}
```

> You can choose from [several Gloo installation options]({{% versioned_link_path fromRoot="/installation/preparation/#deployment-requirements" %}}). This guide assumes you deploy Gloo into a Kubernetes cluster that runs locally in [kind]({{% versioned_link_path fromRoot="/installation/platform_configuration/cluster_setup/#kind" %}}).

Reviewer: ```suggestion
You can choose from [several Gloo Edge installation options]({{% versioned_link_path fromRoot="/installation/preparation/#deployment-requirements" %}}). This guide assumes you deploy Gloo Edge into a Kubernetes cluster that runs locally in [kind]({{% versioned_link_path fromRoot="/installation/platform_configuration/cluster_setup/#kind" %}}).
```
